src/query/storages/fuse/src/operations/mutation/recluster_mutator.rs

```diff
@@ -163,17 +171,9 @@ impl ReclusterMutator {
 
             // find the max point, gather the blocks.
             let mut selected_idx = HashSet::new();
-            let mut find = false;
-            for overlap in point_overlaps {
-                if overlap.len() == max_depth {
-                    overlap.iter().for_each(|&idx| {
-                        selected_idx.insert(idx);
-                    });
-                    find = true;
-                } else if find {
-                    break;
-                }
-            }
+            point_overlaps[max_points[0]].iter().for_each(|idx| {
```

Comment:
just gather the first one? so other blocks with the same max point depth should be thrown away??